src/platform/openiotsdk/KVPsaPsStore.h

```diff
@@ -0,0 +1,196 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Open IoT SDK key-value storage base on flash TDBStore.
+ */
+
+#pragma once
+
+#include <functional>
+#include <string.h>
+
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <platform/PersistedStorage.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+#include <platform/internal/testing/ConfigUnitTest.h>
+
+#include <psa/protected_storage.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+using namespace chip::Platform::PersistedStorage;
+
+// PSA Protected Storage interface provides 64-bit ID number,
+// For Matter we reserved region ranges from 0x00010000 to 0x0001FFFF
+// e.g. key = 0x010214
+// '01' = Matter region
+// '02' = the sub region group base offset (Factory, Config, Counter or KVS)
+// '14' = the id offset inside the group.
+constexpr Key kMatterPsaPaKeyRegion = 0x010000U;
+constexpr inline Key GetPsaPaKey(Key group, uint8_t id)
+{
+    return kMatterPsaPaKeyRegion | (group) << 8 | id;
+}
+
+#define KVS_MAX_ENTRIES 30
+
+#define KVS_KEY_MAP_UPDATE_DELAY_SEC 5
```

Comment:
I think we will have to make this configurable at some point but we are not there yet.